sklearn/semi_supervised/label_propagation.py

```diff
@@ -272,8 +278,9 @@ class LabelPropagation(BaseLabelPropagation):
 
     Parameters
     ----------
-    kernel : {'knn', 'rbf'}
-        String identifier for kernel function to use.
+    kernel : {'knn', 'rbf', function}
+        String identifier for kernel function to use or the kernel function to calculate
+        the [n_samples, n_samples] sized weight matrix.
```

Comment:
I have changed the documentation to talk about the input/output of the kernel function. If there is interest in having this functionality, I will fix the documentation for the `LabelSpread` and `BaseLabelPropagation` class as well for completeness. However, I will give coming up with a convincing test case for #5774 more priority.